### PR TITLE
feat(Tab): add block variant

### PR DESCRIPTION
**Pull request by slin12**
### Overview

Adds new variant of Tabs, to be used in the LE

### PR Checklist

- [ ] Related to designs: https://www.figma.com/file/p6E6kwqWcuF5za0CRn2m4n/Projects-Reskin?node-id=42%3A615
- [ ] Related to JIRA ticket: [PX-812](https://codecademy.atlassian.net/browse/PX-812)
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/30638
and specific places where tabs are currently used (should have no changes vs prod):
https://pr-30638-monolith.dev-eks.codecademy.com/learn/learn-java
https://pr-30638-monolith.dev-eks.codecademy.com/search?query=Python

See new changes here:
https://627568cbae4bd524c881807f--gamut-preview.netlify.app/?path=/docs/molecules-tabs--tabs (scroll down to LE Variant section)

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.a2ff3c.0<br/>@codecademy/gamut-labs@31.11.3-alpha.a2ff3c.0<br/>@codecademy/gamut@46.2.0-alpha.a2ff3c.0<br/>@codecademy/monaco@0.6.37-alpha.a2ff3c.0<br/>@codecademy/styleguide@56.5.0-alpha.a2ff3c.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.4aa916.0<br/>@codecademy/gamut-labs@31.11.3-alpha.4aa916.0<br/>@codecademy/gamut@46.2.0-alpha.4aa916.0<br/>@codecademy/monaco@0.6.37-alpha.4aa916.0<br/>@codecademy/styleguide@56.5.0-alpha.4aa916.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62682f6039ed6f0075f269e2--gamut-preview.netlify.app>https://62682f6039ed6f0075f269e2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62682f6039ed6f0075f269e2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.116-alpha.057a23.0<br/>@codecademy/gamut-labs@31.11.5-alpha.057a23.0<br/>@codecademy/gamut@47.1.0-alpha.057a23.0<br/>@codecademy/monaco@0.6.39-alpha.057a23.0<br/>@codecademy/styleguide@57.1.0-alpha.057a23.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://626975133c794207efd77232--gamut-preview.netlify.app>https://626975133c794207efd77232--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/626975133c794207efd77232>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.116-alpha.80c2eb.0<br/>@codecademy/gamut-labs@31.11.5-alpha.80c2eb.0<br/>@codecademy/gamut@47.1.0-alpha.80c2eb.0<br/>@codecademy/monaco@0.6.39-alpha.80c2eb.0<br/>@codecademy/styleguide@57.1.0-alpha.80c2eb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://626be5c6f4dbba0da097cbef--gamut-preview.netlify.app>https://626be5c6f4dbba0da097cbef--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/626be5c6f4dbba0da097cbef>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.ad7780.0<br/>@codecademy/gamut-labs@31.11.3-alpha.ad7780.0<br/>@codecademy/gamut@46.2.0-alpha.ad7780.0<br/>@codecademy/monaco@0.6.37-alpha.ad7780.0<br/>@codecademy/styleguide@56.5.0-alpha.ad7780.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62755278ae4bd5106d8181ff--gamut-preview.netlify.app>https://62755278ae4bd5106d8181ff--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62755278ae4bd5106d8181ff>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.f0ffd1.0<br/>@codecademy/gamut-labs@31.11.3-alpha.f0ffd1.0<br/>@codecademy/gamut@46.2.0-alpha.f0ffd1.0<br/>@codecademy/monaco@0.6.37-alpha.f0ffd1.0<br/>@codecademy/styleguide@56.5.0-alpha.f0ffd1.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.598de6.0<br/>@codecademy/gamut-labs@31.11.3-alpha.598de6.0<br/>@codecademy/gamut@46.2.0-alpha.598de6.0<br/>@codecademy/monaco@0.6.37-alpha.598de6.0<br/>@codecademy/styleguide@56.5.0-alpha.598de6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627557bdf0a59813dd02d5d3--gamut-preview.netlify.app>https://627557bdf0a59813dd02d5d3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627557bdf0a59813dd02d5d3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.122-alpha.9ed705.0<br/>@codecademy/gamut-labs@31.11.11-alpha.9ed705.0<br/>@codecademy/gamut@47.2.0-alpha.9ed705.0<br/>@codecademy/monaco@0.6.41-alpha.9ed705.0<br/>@codecademy/styleguide@57.1.0-alpha.9ed705.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627568cbae4bd524c881807f--gamut-preview.netlify.app>https://627568cbae4bd524c881807f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627568cbae4bd524c881807f>Deploy Logs</a></p>

File: packages/gamut/src/Tabs/TabProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { createContext, useContext, useMemo } from 'react';
+
+export interface TabContextProps {
+  variant: 'standard' | 'learningEnvironment';
```
Comment (slin12) [inline packages/gamut/src/Tabs/TabProvider.tsx:4]: thoughts on the naming here? is `learningEnvironment` a good name, or should we go with something like `slim`?

File: packages/gamut/src/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -8,31 +13,43 @@ import {
...
-      as={TabsBase}
-      position="relative"
-      zIndex={0}
-      keyboardActivation={TabsKeyboardActivation.Manual}
-      {...props}
-    />
+    <TabProvider value={{ variant: props.variant || 'standard' }}>
+      {/* currently only supporting dark mode for the LE variant */}
+      <ColorMode
+        mode={props.variant === 'learningEnvironment' ? 'dark' : currentMode}
```
Comment (slin12) [inline packages/gamut/src/Tabs/Tabs.tsx:43]: the new tab variant works natively in light mode, but we may want to rethink the colors, etc before allowing it be shipped/available to everyone:
Screen Shot 2022-05-06 at 3 29 38 PM

File: packages/gamut/src/Tabs/TabButton.tsx [hunk 3]
```diff
@@ -22,39 +22,56 @@ const tabSelectedStyles = {
...
-  font: 'inherit',
-  cursor: 'pointer',
-  zIndex: 1,
-  [TabSelectors.HOVER]: {
-    bg: 'background-selected',
-  },
-  [TabSelectors.DISABLED]: {
-    opacity: 0.25,
-    cursor: 'default',
+const tabVariants = variant({
```
Comment (slin12) [inline packages/gamut/src/Tabs/TabButton.tsx:25]: TODO; check w/ Mat on if we need a better 'active' state for the new variant

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.b22be4.0<br/>@codecademy/gamut-labs@31.11.3-alpha.b22be4.0<br/>@codecademy/gamut@46.2.0-alpha.b22be4.0<br/>@codecademy/monaco@0.6.37-alpha.b22be4.0<br/>@codecademy/styleguide@56.5.0-alpha.b22be4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62757a578d4d1234a3d08702--gamut-preview.netlify.app>https://62757a578d4d1234a3d08702--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62757a578d4d1234a3d08702>Deploy Logs</a></p>

File: packages/gamut/src/Tabs/Tabs.tsx (hunk 2 above)
Comment (dreamwasp) [reply packages/gamut/src/Tabs/Tabs.tsx:43]: also having ColorMode defined here was causing some height weirdness with Kip's implementation - could you add this alpha on a branch [there](https://github.com/codecademy-engineering/Codecademy/tree/nd-tj-kp-spring-hackathon-2022)?

File: packages/gamut/src/Tabs/TabProvider.tsx (hunk 1 above)
Comment (dreamwasp) [reply packages/gamut/src/Tabs/TabProvider.tsx:4]: ^^ @songstephen + i think `block` would work here (+ matches our list naming!)

File: packages/gamut/src/Tabs/TabButton.tsx (hunk 3 above)
Comment (slin12) [reply packages/gamut/src/Tabs/TabButton.tsx:25]: checked w/ Stephen here and we're giving all the non selected tabs the same background color (black) in this case so the selected tab stands out more!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.574b09.0<br/>@codecademy/gamut-labs@31.11.3-alpha.574b09.0<br/>@codecademy/gamut@46.2.0-alpha.574b09.0<br/>@codecademy/monaco@0.6.37-alpha.574b09.0<br/>@codecademy/styleguide@56.5.0-alpha.574b09.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6279866b70c49214230dfdc9--gamut-preview.netlify.app>https://6279866b70c49214230dfdc9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6279866b70c49214230dfdc9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.150aac.0<br/>@codecademy/gamut-labs@31.11.3-alpha.150aac.0<br/>@codecademy/gamut@46.2.0-alpha.150aac.0<br/>@codecademy/monaco@0.6.37-alpha.150aac.0<br/>@codecademy/styleguide@56.5.0-alpha.150aac.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627a8f0fb5a19c4326b4b3bf--gamut-preview.netlify.app>https://627a8f0fb5a19c4326b4b3bf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627a8f0fb5a19c4326b4b3bf>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.81a692.0<br/>@codecademy/gamut-labs@31.11.3-alpha.81a692.0<br/>@codecademy/gamut@46.2.0-alpha.81a692.0<br/>@codecademy/monaco@0.6.37-alpha.81a692.0<br/>@codecademy/styleguide@56.5.0-alpha.81a692.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627acfe0a8bd4125abaf41d3--gamut-preview.netlify.app>https://627acfe0a8bd4125abaf41d3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627acfe0a8bd4125abaf41d3>Deploy Logs</a></p>

File: packages/gamut/src/Tabs/Tabs.tsx (hunk 2 above)
Comment (slin12) [reply packages/gamut/src/Tabs/Tabs.tsx]: thanks for catching, cass! I updated their branch to use this newest alpha and i think it's working as expected now!

**Review by dreamwasp — APPROVED** [review]
have like the tiniest thing i didn't catch before but :fire: :fire: :fire:

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.d629a9.0<br/>@codecademy/gamut-labs@31.11.3-alpha.d629a9.0<br/>@codecademy/gamut@46.2.0-alpha.d629a9.0<br/>@codecademy/monaco@0.6.37-alpha.d629a9.0<br/>@codecademy/styleguide@56.5.0-alpha.d629a9.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.114-alpha.431c3e.0<br/>@codecademy/gamut-labs@31.11.3-alpha.431c3e.0<br/>@codecademy/gamut@46.2.0-alpha.431c3e.0<br/>@codecademy/monaco@0.6.37-alpha.431c3e.0<br/>@codecademy/styleguide@56.5.0-alpha.431c3e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627c1f76d1af1d04303ffec8--gamut-preview.netlify.app>https://627c1f76d1af1d04303ffec8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627c1f76d1af1d04303ffec8>Deploy Logs</a></p>

**Pull request merged**